packages/exceptions/src/utils/maps.ts

```diff
@@ -1,88 +1,57 @@
-import { chainErrorMessagesMap, chainCodeErrorMessagesMap } from '../messages'
-import {
-  ErrorContextCode,
-  UnspecifiedErrorCode,
-  ChainCosmosErrorCode,
-} from '../types'
-
-/* todo: remove this with the next chain upgrade */
-export const parseMessage = (message: string) => {
-  const firstParse = message.split('message index: 0:')
-
-  if (firstParse.length === 1) {
-    const [firstParseString] = firstParse
-    const secondParse = firstParseString.split(': invalid request')
-    const [secondParseString] = secondParse
-
-    return secondParseString.trim().trimEnd()
-  }
-
-  const [, firstParseString] = firstParse
-  const [actualMessage] = firstParseString.split(': invalid request')
-
-  return actualMessage.trim().trimEnd()
-}
+import { chainModuleCodeErrorMessagesMap } from '../messages'
+import { ErrorContextCode, UnspecifiedErrorCode } from '../types'
 
-/* todo: remove this with the next chain upgrade */
-const mapMessageByContent = (
-  message: string,
-): { message: string; code: ErrorContextCode } => {
-  const parsedMessage = parseMessage(message)
+const getABCICode = (message: string): number | undefined => {
+  const chainCodePattern = /{key:"ABCICode" value:"(.*?)"}/g
 
-  const messageInMapKey = (
-    Object.keys(chainErrorMessagesMap) as Array<
-      keyof typeof chainErrorMessagesMap
-    >
-  ).find((key) => parsedMessage.toLowerCase().includes(key.toLowerCase()))
+  const chainCode = chainCodePattern.exec(message)
 
-  if (!messageInMapKey) {
-    return { message: parsedMessage, code: UnspecifiedErrorCode }
+  if (!chainCode || chainCode.length < 2) {
+    return
   }
 
-  return chainErrorMessagesMap[messageInMapKey]
+  return Number(chainCode[1])
 }
 
-const getABCICode = (message: string): number | undefined => {
-  const chainCodePattern = /{key:"ABCICode" (.*?)}/g
-  const numericPattern = /\d+/g
+const getContextModule = (message: string): string | undefined => {
+  const chainModulePattern = /{key:"Codespace" value:"(.*?)"}/g
 
-  const chainCodeMsg = message.match(chainCodePattern)
+  const chainCode = chainModulePattern.exec(message)
 
-  if (!chainCodeMsg || chainCodeMsg.length === 0) {
+  if (!chainCode || chainCode.length < 2) {
     return
   }
 
-  const chainCode = chainCodeMsg[0].match(numericPattern)
-
-  if (!chainCode || chainCode.length === 0) {
-    return
-  }
-
-  return Number(chainCode[0])
+  return chainCode[1]
 }
 
 export const mapFailedTransactionMessage = (
   message: string,
-): { message: string; code: ErrorContextCode } => {
+): { message: string; code: ErrorContextCode; module?: string } => {
   const ABCICode = getABCICode(message)
+  const module = getContextModule(message)
+
+  const defaultResponse = {
+    message: 'The request is not known',
```

Comment:
Let's change this to something better, maybe `Transaction execution has failed for unknown reason` or something similar